pkg/operators/ebpf/ebpf.go

```diff
@@ -335,14 +335,24 @@ func (i *ebpfInstance) register(gadgetCtx operators.GadgetContext) error {
 			return fmt.Errorf("adding datasource: %w", err)
 		}
 
-		// TODO: need a link to find out if this is a snapshotter for network; if so, we can add the netns id
-		m.netns, err = ds.AddField(
-			"netns",
-			api.Kind_Uint64,
-			datasource.WithTags("type:gadget_netns_id", "name:netns"),
-		)
-		if err != nil {
-			return fmt.Errorf("adding netnsid")
+		// Add netns field only if the snapshotter will run an iterator attached to a network hook
+		hasNetIter := false
+		for pName := range m.iterators {
+			p, _ := i.collectionSpec.Programs[pName]
+			if p.AttachTo == "tcp" || p.AttachTo == "udp" {
```

Comment:
pkg/operators/ebpf/snapshotter.go also gas some special tests for the iterator kind:
```
case "task":
...
case "tcp", "udp":
```

And I am adding more iterator kinds: https://github.com/inspektor-gadget/inspektor-gadget/pull/2839

In the future, we might add support for "unix", which is also an iterator kind working per netns:
```
$ git grep -w ^DEFINE_BPF_ITER_FUNC
kernel/bpf/cgroup_iter.c:272:DEFINE_BPF_ITER_FUNC(cgroup, struct bpf_iter_meta *meta,
kernel/bpf/link_iter.c:42:DEFINE_BPF_ITER_FUNC(bpf_link, struct bpf_iter_meta *meta, struct bpf_link *link)
kernel/bpf/map_iter.c:42:DEFINE_BPF_ITER_FUNC(bpf_map, struct bpf_iter_meta *meta, struct bpf_map *map)
kernel/bpf/map_iter.c:164:DEFINE_BPF_ITER_FUNC(bpf_map_elem, struct bpf_iter_meta *meta,
kernel/bpf/prog_iter.c:42:DEFINE_BPF_ITER_FUNC(bpf_prog, struct bpf_iter_meta *meta, struct bpf_prog *prog)
kernel/bpf/task_iter.c:174:DEFINE_BPF_ITER_FUNC(task, struct bpf_iter_meta *meta, struct task_struct *task)
kernel/bpf/task_iter.c:346:DEFINE_BPF_ITER_FUNC(task_file, struct bpf_iter_meta *meta,
kernel/bpf/task_iter.c:610:DEFINE_BPF_ITER_FUNC(task_vma, struct bpf_iter_meta *meta,
kernel/kallsyms.c:871:DEFINE_BPF_ITER_FUNC(ksym, struct bpf_iter_meta *meta, struct kallsym_iter *ksym)
net/core/bpf_sk_storage.c:799:DEFINE_BPF_ITER_FUNC(bpf_sk_storage_map, struct bpf_iter_meta *meta,
net/core/sock_map.c:707:DEFINE_BPF_ITER_FUNC(sockmap, struct bpf_iter_meta *meta,
net/ipv4/tcp_ipv4.c:3526:DEFINE_BPF_ITER_FUNC(tcp, struct bpf_iter_meta *meta,
net/ipv4/udp.c:3557:DEFINE_BPF_ITER_FUNC(udp, struct bpf_iter_meta *meta,
net/ipv6/route.c:6650:DEFINE_BPF_ITER_FUNC(ipv6_route, struct bpf_iter_meta *meta, struct fib6_info *rt)
net/netlink/af_netlink.c:2735:DEFINE_BPF_ITER_FUNC(netlink, struct bpf_iter_meta *meta, struct netlink_sock *sk)
net/unix/af_unix.c:3657:DEFINE_BPF_ITER_FUNC(unix, struct bpf_iter_meta *meta,
```

So I am thinking we should get a single function:
```
func isIteratorKindPerNetNs(kind string) bool {
  if kind == "tcp" || kind == "udp" {
    return true
  }
  return false
}
```
So the logic does not get duplicated.

What do you think?

Reply:
Definitely! Thanks for the suggestion. Applied.